filestore/protoboards.go

```diff
@@ -0,0 +1,118 @@
+package filestore
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/influxdata/chronograf"
+)
+
+// ProtoboardExt is the the file extension searched for in the directory for protoboard files
+const ProtoboardExt = ".json"
+
+// Protoboards are instantiable JSON representation of dashbards.  Implements ProtoboardsStore.
+type Protoboards struct {
+	Dir     string                                      // Dir is the directory containing protoboard json definitions
+	Load    func(string) (chronograf.Protoboard, error) // Load receives filename, returns a Protoboard from json file
+	ReadDir func(dirname string) ([]os.FileInfo, error) // ReadDir reads the directory named by dirname and returns a list of directory entries sorted by filename.
+	IDs     chronograf.ID                               // ID generates unique ids for new protoboards
+	Logger  chronograf.Logger
+}
+
+// NewProtoboards constructs a protoboard store wrapping a file system directory
+func NewProtoboards(dir string, ids chronograf.ID, logger chronograf.Logger) chronograf.ProtoboardsStore {
```

Comment:
What do you think about calling this `NewProtoboardStore`?

Reply:
copied from layouts. happy to modernize.